### ISSUE TITLE
Role should override the element style

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor-pdf#1216]
Let's say I have defined the following style on literal elements (`<code>`):

```yml
literal:
    font:
        family: Consolas
        color: #000
        size: 10.5pt
        style: normal
```

Now, I want to apply a custom style when the role "variable" is defined on a literal element:

```yml
role_variable:
    font:
        family: Consolas
        size: 9.0pt
        color: #ffa500
        style: bold
```

If I define the following element in my document, then the font color, size and style defined on the "variable" role are not applied:

```adoc
[.variable]`foo`
```

<img width="60" alt="Capture d’écran 2019-08-16 à 14 24 16" src="https://user-images.githubusercontent.com/333276/63167546-ecc86f80-c031-11e9-928b-9e6f7291730e.png">

As you can see the font color is still black, the font size is 10.5pt and the font weight is normal (ie. the default style is still applied).

As a workaround, it's possible to use a span element which does not have a default/built-in style:

```adoc
[.variable]+foo+
```

<img width="60" alt="Capture d’écran 2019-08-16 à 14 26 54" src="https://user-images.githubusercontent.com/333276/63167590-08337a80-c032-11e9-8594-d1205d6d4412.png">

Here the font color, style and size are correctly applied.

I think that the style defined on a role should override the style defined on the element (ie. has a higher precedence).

**Comment by mojavelinux** [comment]
You're correct, that's how it should work. And I think it's possible to accommodate that behavior.

The situation is complicated by the fact that the precedence of formatting is backwards in Asciidoctor PDF. The formatting should be applied from outside in, but is currently applied from the inside out. This is something that needs to be fixed. See #1219. We may still be able to fix this behavior in the interim.

**Comment by mojavelinux** [comment]
It can be fixed for the simple case:

```
[.yellow]`yellow`
```

However, it then fails for the nested case:

```
[.green]#green [.yellow]`yellow` green#
```

Ultimately, we have to fix #1219 in order to be able to do this.
